### DISCUSSION TITLE
TypeError: Cannot assign to read only property '<propertyName>' of object '#<Object>'

### DISCUSSION THREAD

**Discussion by zjsng** [react-hook-form/react-hook-form#10905] [Q&A]
I'm at a loss, I've been trying to debug this for hours and can't seem to fix it for some reason.

I am using cloneDeep from Lodash to reset, and at least now it only occurs on two fields. As well as this following error: 

`TypeError: Cannot assign to read only property '0' of object '[object Array]'.`

Any ideas how I can even attempt to debug this?

**Comment by Moshyfawn** [comment]
#3715

**Reply by tmnrp** [reply]
Did you find the root cause/ fix ?
i have the same error when performing useFieldArray operation on nessed react hook form zod schema

**Comment by tankotun** [comment]
If you use useFieldArray.update to update an object in the array, it may eventually cause this error. You should use setValue from useFormContext to update.

**Reply by GEMI** [reply]
Thanks so much! I’ve been digging into this for hours, and you just saved me.
